solc/CommandLineInterface.cpp

```diff
@@ -1152,14 +1149,24 @@ void CommandLineInterface::outputCompilationResults()
 		{
 			if (m_args.count(g_argOutputDir))
 			{
-				stringstream data;
-				m_compiler->streamAssembly(data, contract, m_sourceCodes, m_args.count(g_argAsmJson));
-				createFile(m_compiler->filesystemFriendlyName(contract) + (m_args.count(g_argAsmJson) ? "_evm.json" : ".evm"), data.str());
+				if (m_args.count(g_argAsmJson)
+				{
+					Json::Value ret = m_compiler->assemblyJSON(contract, m_sourceCodes);
+					createFile(m_compiler->filesystemFriendlyName(contract) + "_evm.json", dev::jsonPrettyPrint(ret));
```

Comment:
Would prefer to reduce code duplication here.

Reply:
Done